editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,378 @@
+      serviceGeodatabase.addDoneLoadingListener(() -> {
+        if (serviceGeodatabase.getLoadStatus() == LoadStatus.LOADED) {
+
+          // when the service geodatabase has loaded get the default version
+          defaultVersionName = serviceGeodatabase.getDefaultVersionName();
+
+          // get the service feature table from the service geodatabase
+          if (serviceGeodatabase.getTable(0) != null) {
+            serviceFeatureTable = serviceGeodatabase.getTable(0);
+
+            // create a feature layer from the service feature table and add it to the map
+            featureLayer = new FeatureLayer(serviceFeatureTable);
+            map.getOperationalLayers().add(featureLayer);
+            featureLayer.addDoneLoadingListener(() -> {
+              if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
+
+                // when the feature layer has loaded set the viewpoint and update the UI
+                mapView.setViewpointAsync(new Viewpoint(featureLayer.getFullExtent()));
+                progressIndicator.setVisible(false);
+                createVersionButton.setDisable(false);
+                currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+              } else showAlert("Feature layer failed to load");
+            });
+          } else showAlert("Unable to get the service feature table");
+        } else {
+          progressIndicator.setVisible(false);
+          showAlert("Service geodatabase failed to load");
+        }
+      });
+
+      // listen to clicks on the map to select or move features
+      mapView.setOnMouseClicked(event -> {
+
+        // create a point from where the user clicked
+        Point2D point = new Point2D(event.getX(), event.getY());
+
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // reset the UI
+          featureLayer.clearSelection();
+          editFeatureVBox.setDisable(true);
+
+          // select the clicked feature
+          selectFeature(point);
+        }
+
+        if (event.isStillSincePress() && event.getButton() == MouseButton.SECONDARY) {
+          // if a feature is selected and the current version is not the default, update the feature's geometry
+          if (selectedFeature != null && !serviceGeodatabase.getVersionName().equals(defaultVersionName)) {
+            Point mapPoint = mapView.screenToLocation(point);
+            selectedFeature.setGeometry(mapPoint);
+            updateFeature(selectedFeature);
+          }
+        }
+      });
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Create a new branch version using user defined values.
+   */
+  @FXML
+  private void handleCreateVersionButtonClicked() {
+
+    // validate version name input
+    String inputName = nameTextField.getText();
+   if (!isTextInputValid(inputName)) {
+     return;
+   };
```

Comment:
getting an IDE prompt to shorten the above to the following:

```suggestion
   if (!isTextInputValid(inputName)) return;
```